platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/ConnectivityManager.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.network.topology;
+
+import static com.swirlds.logging.legacy.LogMarker.EXCEPTION;
+import static com.swirlds.logging.legacy.LogMarker.NETWORK;
+
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.network.Connection;
+import com.swirlds.platform.network.ConnectionManager;
+import com.swirlds.platform.network.InboundConnectionManager;
+import com.swirlds.platform.network.OutboundConnectionManager;
+import com.swirlds.platform.network.PeerInfo;
+import com.swirlds.platform.network.connectivity.OutboundConnectionCreator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Pre-builds connection managers for the supplied topology, does not allow changes at runtime
+ */
+public class ConnectivityManager {
+    private static final Logger logger = LogManager.getLogger(ConnectivityManager.class);
+    private final NetworkTopology topology;
+    private final OutboundConnectionCreator connectionCreator;
+    private Map<ConnectionMapping, ConnectionManager> connectionManagers;
+
+    /** holds the nodeIds of connection managers for fast lookup*/
+    private final Set<NodeId> nodeIdsOfConnectionManagers = new HashSet<>();
+
+    /**
+     * Create a new connectivity manager with the given topology and connection creator
+     *
+     * @param topology          the network topology
+     * @param connectionCreator the connection creator
+     */
+    public ConnectivityManager(
+            @NonNull final NetworkTopology topology, @NonNull final OutboundConnectionCreator connectionCreator) {
+        this.topology = Objects.requireNonNull(topology);
+        this.connectionCreator = Objects.requireNonNull(connectionCreator);
+
+        this.connectionManagers = createConnectionManagers();
```

Comment:
there are 2 pathways for creating connection managers, one based on topology and the other based on the peer list. i think it would be a lot simpler if there was only 1